src/frontend/src/lib/ic-pub-key/src/cli.ts

```diff
@@ -68,4 +70,25 @@ export const deriveBtcAddress = async (user: string, network: BitcoinNetwork): P
 	return btc_address;
 };
 
+/* istanbul ignore next */
+export const deriveSolAddress = async (user: string, derivationPath: string[]): Promise<string> => {
+	const pubkey = '476374d9df3a8af28d3164dc2422cff894482eadd1295290b6d9ad92b2eeaa5c';
+	const chaincode = '0000000000000000000000000000000000000000000000000000000000000000';
+
+	const principal = Principal.fromText(user);
+
+	let derivationPathObj = new DerivationPath([
+		Uint8Array.from([0, 0, 0, 0, 2, 48, 0, 113, 1, 1]),
```

Comment:
This is the chain fusion signer canister ID.  If you have that as a variable, which you probably do as you have to make API calls to the signer, this can be made to work on staging and prod alike.  On staging we also use the test key, the moment, so in my opinion this is not worth doing in this PR.